internal/core/encoding.go

```diff
@@ -110,6 +128,19 @@ func NormalizeFieldValue(fieldDesc client.FieldDefinition, val any) (any, error)
 			case uint:
 				return float64(v), nil
 			}
+		case client.FieldKind_NILLABLE_FLOAT32:
+			switch v := val.(type) {
```

Comment:
todo: Code-cov is flagging a lot of this as untested.  Is it dead code?  Will declaring an array `[1, 1.1]` not result in at least one of the `int` cases being hit?

Please include a test where some of the values lack decimal points if there are no such tests already.

If there are such tests, I *suggest* the deletion of the dead code branches.

Same todo/suggestion applies to `convertToFloat32` too.

Reply:
This function is used on unmarshalling from CBOR. As a result, I think a lot of this is defensive programming. I see this todo as out of scope of this PR. We can open an issue to review this at some point if you'd like though.

Reply:
I'm not sure I understand why adding an integration test with an array `[1, 1.1]` or similar would be out of scope as it would be covering behaviour introduced in this PR.

Or are you saying such a test would not cover that case?

Reply:
I think it's out of scope because it is not new behaviour. It's just a new case in the switch for a different type but the contents of the case are the same as the others which are also not covered. To cover all the possibilities, we would have to figure out what CBOR converts the values as when marshalling. So I'm not sur that the test you propose would cover much more.